doc/nyan/engine_types.md

```diff
@@ -0,0 +1,191 @@
+
+AnimatedBuilding(Building, Animation):
+	# Used for buildings that consist of a static frame and an animation on top.
+
+	animation : orderedset(file)
+
+
+Ability(): # WIP
+	# Every unit (WIP: or even buildings, see 'CarryAbility',castles... ), that can do something, has an 'Ability'.
+	# WIP: what members belong here?
+
+HUDAbility(Ability, HUDElement): # WIP
+	# Creates a button on the left of the HUD. For Trebuchet packing and
+	# unpacking, unloading units or relics. Maybe even unit patrol andformations.
+
+MoveAbility(Ability, Animation):
+	# You want to move? Use this class.
+
+	speed : float
+	instant : bool = False # WIP: how about checking (speed < 0) ⇒ instant before
+	                       # path searching?
+	range : float
+
+CooldownAbility(Ability):
+	# Any executed 'Ability', that inherits from this, can only be executed again
+	# after recharge_time in seconds.
+	recharge_time : float
+
+GatherAbility(Ability, Animation): # WIP
+	# Gathering resources requires this Ability
+
+	allowedResource : set(Resource) # Allowed ressources?
+	maxAmount : int # maximum amount  that can be carried
+	speed : float # gathered units per second
+
+	# WIP2: proposal: allow different maximum amounts for different resources:
+
+#	allowedResource : orderedset(Resource)
+#	maxAmount : orderedSet(int) # does <10,5,0,10> work? it is a set after all.
+
+	# WIP3: if WIP2 is not possible: implement an 'Ability' that allows to gather only one resource.
+	# A worker would then need 4 gather abilities added to it's 'abilities' set.
+
+CarryAbility(Ability): # WIP: find a better name?
+	# Used by priests for relics and transport ship (both indirectly [animation needed]) and by castles etc. for many types of units.
+	# WIP1: how should this look like? keep it general?
+	# WIP3: HUDElement for releasing every contained unit and setting a meeting point needed. (But not for priests. -> Check for MovableUnit in allowedUnits?
+	#       or use the alternative below for everything but relics.)
+
+	allowedUnits : set(UnitTypes) # what can be collected?
+	maxUnits : int
+	maxUnitsize : int # maximum size of each unit. If we want this, every movable unit should have a size member declared by a base class.
+	tooDamagedRate : float # percentage of health at which no units can be contained any longer.
+
+# Alternative:
+Shelter(Ability):
+	# A 'Unit' which might hold other units. Used by Castle, Towncenter, TransporterShip, but not priest.
+
+	allowedUnits : set(UnitTypes) # what can be collected?
+	maxUnits : int
+	maxUnitsize : int # maximum size of each unit. If we want this, every movable unit should have a size member declared by a base class.
+	collapsingRate : float # percentage of health at which no units can be contained any longer.
+
+AttackAbility(Ability):
+	# Every attack shall inherit from this.
+	# WIP: how will damage calculation work in oa?
+
+	range : float
+	rate : float
+	damage : float # rename to baseDamage ?
+
+
+HealingAbility(Ability): #WIP
```

Comment:
Probably must be split. `Healing`, `Repairing` and `Healing_inside_a_building` must be separate because its have different logic behind it (eg. repair has cost).

`Healing_inside_a_building` could also be megred with `Garison`.